### ISSUE TITLE
Problems on RPi400

### ISSUE THREAD

**Issue by pa3byv** [dj0abr/QO100_Transceiver#1]
I was pleasantly surprised by the announcement of the Linux Transceiver. Thought it would be a welcome addition to my arsenal. Installed it according to the description on an Raspberry Pi 400 with a PiSDR card. It started with the opening screen, but didn't get any further after doing the configuration. Redid it on another SD card with latest Buster installed, same problem, but now I got as far as a spectrum and waterfall with multiple symmetrical peaks, updating once every two seconds. Burned a new SD card with 21-03-04 Buster, again same result. Used two different Pluto's (both rev B), and USB-Dongle. Audio loop worked, although with an annoying hum, RF loop didn't. Also the settings for the frequencies were lost every time at restart of the app. On install I didn't get any errors, only warnings about Jack daemon not installed. At startup also warnings that Jack server didn't startup. But Alsa worked, so probably shouldn't be a problem Any idea what might be wrong?

73
Poll
![Linux trx](https://user-images.githubusercontent.com/85561563/121186534-7c728a00-c867-11eb-95a8-308308862d7b.jpg)

**Comment by dj0abr** [comment]
Hello Poll,

I'm not sure what the PI400 is, but what I read it is a PI-4 with a keyboard. So it should be compatible.

if the Screen updates only every 2 seconds, there could be a problem with system performance, usually it updates multiple times per second.

The Jack-warning is normal, I can see it too, but no problem.

* Could you please send me a screenshot of "htop" when the software is running, and
* your config file, which is loacted in the home folder /home/pi and has the name: trx.cfg
* what OS did you install? I know its working fine with Raspi-OS 32 or 64 but, but does not work with the Ubuntu version offered for the Raspberry (all get very, very slow).

vy 73
Kurt

**Comment by pa3byv** [comment]
Dear Kurt,

Thank you for replying, and thank you for writing this software. I know it must have taken a lot of your time and expertise. 

Today I tried again, and had the same failures and also (forgot to report earlier) buffer full warnings, as you can see in the accompanying pictures and files.


But, I also tried a Pluto+, which did work! So the problem seems to be with both of my Pluto rev.B’s.

The Pluto’s run different fw, but all with F5OEO fw on it. One Pluto runs Perseverance 2202, one runs 2808 and I believe the Pluto+ is also running on 2808.

So, with the Pluto+ running you can see, that CPU use is actually pretty moderate, below 50% on all cores. With the Pluto-B, that is not running well, the cores run very low, so the software seems not to run its normal speed. Also, that is when you see the strange peaks and the 2-second update interval.

All three Pluto’s work correctly in normal operation. The first Pluto-B, also used in this test, is in daily use on QO-100 in SSB with SDRConsole.

The other Pluto-B and the Pluto+ are in daily use for DATV on the WB-TP, directly from a PC or from the Portsdown 4.

The RPI400 is indeed a RPi4, 4GB, on a larger flatter board to fit in a keyboard. It has one USB-2 less and and the 3.5 mm V/A connector is absent. Otherwise the same thing.

I hope the files and pictures give you enough information, otherwise please feel free to ask for more info. As the picture files are to big I will be sending them in two emails.

73

Poll, PA3BYV

**Comment by pa3byv** [comment]
Second zip

**Comment by pa3byv** [comment]
And to answer your last question: 2021-03-04-raspios-buster-armhf.

73
Poll

**Comment by dj0abr** [comment]
I'm not sure what happens on your system.

So I did it from scratch using your image. I have documented each single step here:
https://wiki.amsat-dl.org/doku.php?id=en:plutotrx:raspi

After this procedure I took an image of the running system, the download link is in above description.

I have tested it with an older pluto Rev.B, and a brand new Rev.C and also with the default firmware as well as the F5OEO firmware. 

(I did not receive the files which should be included in your emails, please send them again, thank you)

**Comment by pa3byv** [comment]
Forwarding again with the same .zip file to see if it is a send or a receive problem.

73,
Poll


Begin forwarded message:

**Comment by pa3byv** [comment]
Dear Kurt,

I have forwarded the same message including the .zip files with a copy to myself and received the zip file with the message. This is the second zip file. Please let me know if you didn't receive it. If so, I probably need to send it to your other email address.

73
Poll


Begin forwarded message:

**Comment by pa3byv** [comment]
Hi Kurt,

I was able to get one of the Pluto-B’s working: First installed the official AD fw version 0.33, but that didn’t cut it. Then I removed all leftover files and directories from previous DATV patches from /mnt/jffs2/ and now it came to life! So, although I’m not sure what exactly is to blame, it certainly isn’t your software. Seems to be an incompatibility between a previous patch and your software. I will try to reinstall the -for me necessary- 2202 F5OEO firmware and see what happens.

Vy 73,
Poll

**Comment by dj0abr** [comment]
sorry Poll, but again no attachment received.
Please try my other email:  dj0abr@darc.de and send from your personal email, not via github.
Thank you,
Kurt

**Comment by dj0abr** [comment]
ok thanks Poll, so no more need for the files.

I have the F5OEO firmware installed because I also stream DATV. This works fine with the transceiver software

vy 73
Kurt

**Comment by pa3byv** [comment]
I have been using multiple versions of the F5OEO software and also versions of patches by F5UII, so all of them could have left something incompatible. On this particular Pluto-B the latest fw was Perseverance 2202.

73
Poll, PA3BYV

**Comment by xtranerd** [comment]
Hi - I was very pleased to see a Linux QO-100 application. I have used Sdrangel on linux to transmit and receive on QO-100 which worked - but in the end I ended up using SDRConsole on Windows as  it supported beacon tracking easily.

I installed QO100_Transceiver on a Debian 10 Buster install on an I3 laptop and the install went OK -  I had to apt install a missing  gtk theme but that was the only issue.

The first problem is that I cant resize (or reduce/expand) the program window and there is only a "X" to close option on the applications window - this means I can't see the bottom row with all buttons on it - I got around this by using ALT+F7 and dragging the window and ten resizing it but this is really inconvenient as bits of the profgramme view are "chopped off"

The other issus are -

The configuration after initial setup and calibration is not saved for the next time the programme is run

I'm using a ADALM-PLUTO (with original firmware) over etherent and receive works-  but I can't transmit (I know it works fine with Sdrangel in linux and SDRConsole in Windows. Audio and rf loopback both work fine.  I think I can just about hear my signal on the downlink but I think the output  rf level from the pluto is far too low . I understand other programmers have had this issue, I have a portable station with 1.8 W and a 60 cm dish so I need as much rf output from the pluto as possible. On SDRConsole the output level is set at about 88% to ensure I don't overdrive the next rf amplifier.

If these issues could be fixed and perhaps some frquency correction for lnb frequency drift  (perhaps tracking the beacons) and a signal meter for alignment this would be a great option for portable QO-100 operation.

Thanks again for all your hard work on this project

James G1PVZ
